From: Christoph Egger <debian@christoph-egger.org>
Subject: [PATCH] debian/arch-support

Building on non-linux architectures currently fails with unpatched
irrlicht because irrlicht tries to create Joystick support using
linux-specific headers. However there's infrastructure to disable
Joystick support, we just need to activate that on non-linux
architectures.

Additionally if built on a sparc machine irrlicht assumes wrongly it's
a solaris system. We fix this wrong assumption as our sparc builds are
all on linux.

Finally irrlicht exceeds the size constraights for -fpic requiring to
build with -fPIC. As upstream doesn't do that we need to fix this for
sparc and s390 builds (powerPC?).

Signed-off-by: Christoph Egger <debian@christoph-egger.org>

---
 irrlicht/include/IrrCompileConfig.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/irrlicht/include/IrrCompileConfig.h
+++ b/irrlicht/include/IrrCompileConfig.h
@@ -112,6 +112,10 @@
 #if !defined(_IRR_WINDOWS_API_) && !defined(_IRR_OSX_PLATFORM_) && !defined(_IRR_IOS_PLATFORM_) && !defined(_IRR_ANDROID_PLATFORM_) && !defined(_IRR_EMSCRIPTEN_PLATFORM_)
 #ifndef _IRR_SOLARIS_PLATFORM_
 #define _IRR_LINUX_PLATFORM_
+#include <endian.h>
+ #if __BYTE_ORDER == __BIG_ENDIAN
+  #define __BIG_ENDIAN__
+ #endif
 #endif
 #define _IRR_POSIX_API_
 #ifndef _IRR_COMPILE_WITH_SDL_DEVICE_
@@ -441,7 +445,7 @@
 
 #endif // _IRR_WINDOWS_API_
 
-#if defined(_IRR_SOLARIS_PLATFORM_)
+#if defined(_IRR_SOLARIS_PLATFORM_) || defined(__FreeBSD_kernel__) || defined(__gnu_hurd__)
 	#undef _IRR_COMPILE_WITH_JOYSTICK_EVENTS_
 #endif
 
